### ISSUE TITLE
Clean up tests/CMakeLists.txt

### ISSUE THREAD

**Issue by jdiez17** [esa/pagmo#115]
Instead of manually adding three lines per unit test, cmake should go through every `test_*.cpp` file and add it to the unit test list automatically.

**Comment by darioizzo** [comment]
Good idea, but there are some tests that need to be compiled only if some of the build options are activated. How would you deal with those?

**Comment by bluescarni** [comment]
Why not writing a small macro to wrap those three lines in one instead?

I am not a fan of auto-discovery or file globbing in these cases. The list of tests (or, a list of source files more generally) is not going to change often, and globbing can make it quite hard to understand what is going on (e.g., if you have temporary .cpp files in there for debug or whatever).

**Comment by jdiez17** [comment]
Good point, globbing might not be the best way. We could have a list of
tests, and the build options that need to be activated to run each test.

**Comment by bluescarni** [comment]
@jdiez17 Just for reference, this is a possible way of doing it:

https://github.com/bluescarni/piranha/blob/master/tests/CMakeLists.txt
